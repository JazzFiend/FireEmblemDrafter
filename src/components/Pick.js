import React, { Component } from "react";
import PropTypes from 'prop-types';

export class Pick extends Component {
  render() {
    return (<div>
      Make a Selection
        {this.buildButtons()}
    </div>);
  }

  buildButtons() {
    let buttonArray = Array(0);
    for (let i = 0; i < this.props.pick.length; i++) {
      buttonArray.push(
        <button onClick={() => this.props.onClick(i)} key={i}>
          {this.props.pick[i]}
        </button>
      );
    }
    return buttonArray;
  }
}

Pick.propTypes = {
  pick: PropTypes.arrayOf(PropTypes.string),
  onClick: PropTypes.func,
};
