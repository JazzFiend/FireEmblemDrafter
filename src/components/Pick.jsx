import React, { Component } from "react";

class Pick extends Component {
  buildButtons(pick, onClick) {
    let buttonArray = Array(0);
    for (let i = 0; i < pick.length; i++) {
      buttonArray.push(
        <button type="button" data-testid="pick-button" onClick={() => onClick(i)} key={i}>
          {pick[i]}
        </button>
      );
    }
    return buttonArray;
  }

  render() {
    let propsCopy = this.props;
    return (
      <div data-testid="pick">
      Make a Selection
        {this.buildButtons(propsCopy.pick, propsCopy.onClick)}
      </div>
    );
  }
}

Pick.defaultProps = {
  pick: [],
  onClick: () => {},
};

export default Pick;
