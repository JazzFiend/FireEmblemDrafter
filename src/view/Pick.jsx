import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Pick extends PureComponent {
  buildButtons() {
    const { pick, onClick } = this.props;
    const buttonArray = Array(0);

    for (let i = 0; i < pick.length; i++) {
      buttonArray.push(
        <button type="button" data-testid="pick-button" onClick={() => onClick(i)} key={i}>
          {pick[i]}
        </button>,
      );
    }
    return buttonArray;
  }

  render() {
    return (
      <div data-testid="pick">
        Make a Selection
        {this.buildButtons()}
      </div>
    );
  }
}

Pick.propTypes = {
  pick: PropTypes.arrayOf(PropTypes.string),
  onClick: PropTypes.func,
};

Pick.defaultProps = {
  pick: [],
  onClick: () => {},
};

export default Pick;
