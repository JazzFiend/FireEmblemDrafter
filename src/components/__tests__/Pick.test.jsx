import _ from 'lodash';
import React from 'react';
import {
  render,
  fireEvent,
  cleanup, } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Pick from '../Pick';

afterEach(cleanup);

test('renders and matches snapshot', () => {
  const { asFragment } = render(<Pick pick={['a', 'b', 'c']} onClick={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

test('contains appropriate buttons', () => {
  const testPick =  ['x', 'y', 'z'];
  const { getAllByTestId } = render(<Pick pick={testPick} onClick={() => {}} />);
  let pickButtons = getAllByTestId('pick-button');

  expect(pickButtons.length).toBe(3);
  _.forEach(pickButtons, function(button, index) {
    expect(button).toHaveTextContent(testPick[index]);
  });
});

test('click runs passed in function', () => {
  const onClick = jest.fn();
  const { getByTestId } = render(<Pick pick={['x']} onClick={(index) => onClick(index)} />);

  fireEvent.click(getByTestId('pick-button'));
  expect(onClick).toHaveBeenCalledWith(0);
});
