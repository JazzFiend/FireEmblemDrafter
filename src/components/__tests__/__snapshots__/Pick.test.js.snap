// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="pick"
  >
    Make a Selection
    <button
      data-testid="pick-button"
    >
      a
    </button>
    <button
      data-testid="pick-button"
    >
      b
    </button>
    <button
      data-testid="pick-button"
    >
      c
    </button>
  </div>
</DocumentFragment>
`;
